packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -975,6 +975,10 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
     context.pushClipRect(needsCompositing, offset, Offset.zero & size, defaultPaint);
 
     assert(() {
+      // There's no point in drawing the overflow indicator if we're empty.
+      if (size.isEmpty)
+        return true;
```

Comment:
As discussed at lunch, we should just do this at the top level (not even attempt to paint children in this case).